lib/Bio/EnsEMBL/DataCheck/Checks/DisplayXref.pm

```diff
@@ -0,0 +1,101 @@
+=head1 LICENSE
+
+Copyright [2018-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::DisplayXref;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'DisplayXref',
+  DESCRIPTION    => 'Check that display_xrefs are set OK',
+  GROUPS         => ['core', 'xref'],
+  DATACHECK_TYPE => 'critical',
+  TABLES         => ['xref'],
+};
+
+sub tests {
+
+  my ($self) = @_;
+
+  my $species_id = $self->dba->species_id;
+  foreach my $type ("gene", "transcript"){
+
+    my $desc_1 = "${type}s  with valid display_xref_id";
```

Comment:
This is not critical - most non-vert dbs do not have display xrefs, and new vertebrates might not either; this test needs to be moved to a separate advisory datacheck.